```
I0224 22:11:58.910] docker push gcr.io/k8s-jkns-pr-kubemark/kubemark:zjlm3i
I0224 22:11:58.986] The push refers to a repository [gcr.io/k8s-jkns-pr-kubemark/kubemark]
I0224 22:11:59.004] c429cafa92a3: Preparing
I0224 22:11:59.004] ffc4c11463ee: Preparing
W0224 22:11:59.198] denied: Unable to access the repository, please check that you have permission to access it.
W0224 22:11:59.200] make: *** [gcloudpush] Error 1
W0224 22:11:59.200] Make attempt 1 failed. Retrying.
I0224 22:11:59.301] Makefile:31: recipe for target 'gcloudpush' failed
```

Looks like legitimate failures. Perhaps the auth provider isn't configured properly in CI?

--- a/build/build-image/cross/Makefile
+++ b/build/build-image/cross/Makefile
@@ -24,4 +24,4 @@ build:
 	docker build --pull -t staging-k8s.gcr.io/$(IMAGE):$(TAG) .
 
 push: build
-	gcloud docker -- push staging-k8s.gcr.io/$(IMAGE):$(TAG)
+	docker push staging-k8s.gcr.io/$(IMAGE):$(TAG)

--- a/build/debian-base/Makefile
+++ b/build/debian-base/Makefile
@@ -71,7 +71,7 @@ endif
 	rm -rf $(TEMP_DIR)
 
 push: build
-	gcloud docker -- push $(REGISTRY)/$(IMAGE)-$(ARCH):$(TAG)
+	docker push $(REGISTRY)/$(IMAGE)-$(ARCH):$(TAG)
 
 clean:
 	docker rmi -f $(REGISTRY)/$(IMAGE)-$(ARCH):$(TAG) || true

--- a/build/debian-hyperkube-base/Makefile
+++ b/build/debian-hyperkube-base/Makefile
@@ -57,4 +57,4 @@ endif
 	rm -rf $(TEMP_DIR)
 
 push: build
-	gcloud docker -- push $(REGISTRY)/$(IMAGE)-$(ARCH):$(TAG)
+	docker push $(REGISTRY)/$(IMAGE)-$(ARCH):$(TAG)

--- a/build/debian-iptables/Makefile
+++ b/build/debian-iptables/Makefile
@@ -55,6 +55,6 @@ endif
 	docker build --pull -t $(REGISTRY)/$(IMAGE)-$(ARCH):$(TAG) $(TEMP_DIR)
 
 push: build
-	gcloud docker -- push $(REGISTRY)/$(IMAGE)-$(ARCH):$(TAG)
+	docker push $(REGISTRY)/$(IMAGE)-$(ARCH):$(TAG)
 
 all: push

--- a/build/pause/Makefile
+++ b/build/pause/Makefile
@@ -91,7 +91,7 @@ container: .container-$(ARCH)
 
 push: .push-$(ARCH)
 .push-$(ARCH): .container-$(ARCH)
-	gcloud docker -- push $(IMAGE_WITH_ARCH):$(TAG)
+	docker push $(IMAGE_WITH_ARCH):$(TAG)
 	touch $@
 
 # Useful for testing, not automatically included in container image

--- a/cluster/addons/addon-manager/Makefile
+++ b/cluster/addons/addon-manager/Makefile
@@ -46,12 +46,12 @@ build:
 	docker build --pull -t $(IMAGE)-$(ARCH):$(VERSION) $(TEMP_DIR)
 
 push: build
-	gcloud docker -- push $(IMAGE)-$(ARCH):$(VERSION)
+	docker push $(IMAGE)-$(ARCH):$(VERSION)
 ifeq ($(ARCH),amd64)
 	# Backward compatibility. TODO: deprecate this image tag
 	docker rmi $(IMAGE):$(VERSION) 2>/dev/null || true
 	docker tag $(IMAGE)-$(ARCH):$(VERSION) $(IMAGE):$(VERSION)
-	gcloud docker -- push $(IMAGE):$(VERSION)
+	docker push $(IMAGE):$(VERSION)
 endif
 
 clean:

--- a/cluster/addons/fluentd-elasticsearch/es-image/Makefile
+++ b/cluster/addons/fluentd-elasticsearch/es-image/Makefile
@@ -22,7 +22,7 @@ build:
 	docker build --pull -t $(PREFIX)/$(IMAGE):$(TAG) .
 
 push:
-	gcloud docker -- push $(PREFIX)/$(IMAGE):$(TAG)
+	docker push $(PREFIX)/$(IMAGE):$(TAG)
 
 binary:
 	CGO_ENABLED=0 GOOS=linux go build -a -ldflags "-w" elasticsearch_logging_discovery.go

--- a/cluster/addons/fluentd-elasticsearch/fluentd-es-image/Makefile
+++ b/cluster/addons/fluentd-elasticsearch/fluentd-es-image/Makefile
@@ -22,4 +22,4 @@ build:
 	docker build --pull -t $(PREFIX)/$(IMAGE):$(TAG) .
 
 push:
-	gcloud docker -- push $(PREFIX)/$(IMAGE):$(TAG)
+	docker push $(PREFIX)/$(IMAGE):$(TAG)

--- a/cluster/addons/python-image/Makefile
+++ b/cluster/addons/python-image/Makefile
@@ -21,5 +21,5 @@ build:
 	docker build --pull -t "$(IMAGE):$(VERSION)" .
 
 push:
-	gcloud docker -- push "$(IMAGE):$(VERSION)"
+	docker push "$(IMAGE):$(VERSION)"
 

--- a/cluster/gce/gci/mounter/Makefile
+++ b/cluster/gce/gci/mounter/Makefile
@@ -22,7 +22,7 @@ container:
 	docker build --pull -t ${REGISTRY}/${IMAGE}:${TAG} .
 
 push: 
-	gcloud docker -- push ${REGISTRY}/${IMAGE}:${TAG}
+	docker push ${REGISTRY}/${IMAGE}:${TAG}
 
 upload:
 	./stage-upload.sh ${TAG} ${REGISTRY}/${IMAGE}:${TAG}

--- a/cluster/images/etcd-empty-dir-cleanup/Makefile
+++ b/cluster/images/etcd-empty-dir-cleanup/Makefile
@@ -29,4 +29,4 @@ build: clean
 	rm -rf etcdctl etcd-v$(ETCD_VERSION)-linux-amd64 etcd-v$(ETCD_VERSION)-linux-amd64.tar.gz
 
 push: build
-	gcloud docker -- push $(IMAGE):$(TAG)
+	docker push $(IMAGE):$(TAG)

--- a/cluster/images/etcd-version-monitor/Makefile
+++ b/cluster/images/etcd-version-monitor/Makefile
@@ -40,7 +40,7 @@ build:
 	docker build -t $(IMAGE) $(TEMP_DIR)
 
 push: build
-	gcloud docker -- push $(IMAGE)
+	docker push $(IMAGE)
 
 all: build
 

--- a/cluster/images/etcd/Makefile
+++ b/cluster/images/etcd/Makefile
@@ -108,12 +108,12 @@ endif
 	docker build --pull -t $(REGISTRY)/etcd-$(ARCH):$(REGISTRY_TAG) $(TEMP_DIR)
 
 push: build
-	gcloud docker -- push $(REGISTRY)/etcd-$(ARCH):$(REGISTRY_TAG)
+	docker push $(REGISTRY)/etcd-$(ARCH):$(REGISTRY_TAG)
 
 ifeq ($(ARCH),amd64)
 	# Backward compatibility. TODO: deprecate this image tag
 	docker tag $(REGISTRY)/etcd-$(ARCH):$(REGISTRY_TAG) $(REGISTRY)/etcd:$(REGISTRY_TAG)
-	gcloud docker -- push $(REGISTRY)/etcd:$(REGISTRY_TAG)
+	docker push $(REGISTRY)/etcd:$(REGISTRY_TAG)
 endif
 
 ETCD2_ROLLBACK_NEW_TAG=3.0.17

--- a/cluster/images/hyperkube/Makefile
+++ b/cluster/images/hyperkube/Makefile
@@ -45,11 +45,11 @@ endif
 	rm -rf "${TEMP_DIR}"
 
 push: build
-	gcloud docker -- push ${REGISTRY}/hyperkube-${ARCH}:${VERSION}
+	docker push ${REGISTRY}/hyperkube-${ARCH}:${VERSION}
 ifeq ($(ARCH),amd64)
 	docker rmi ${REGISTRY}/hyperkube:${VERSION} 2>/dev/null || true
 	docker tag ${REGISTRY}/hyperkube-${ARCH}:${VERSION} ${REGISTRY}/hyperkube:${VERSION}
-	gcloud docker -- push ${REGISTRY}/hyperkube:${VERSION}
+	docker push ${REGISTRY}/hyperkube:${VERSION}
 endif
 
 .PHONY: build push all

--- a/cluster/images/kubemark/Makefile
+++ b/cluster/images/kubemark/Makefile
@@ -28,7 +28,7 @@ build:
 	docker build --pull -t $(REGISTRY)/kubemark:$(IMAGE_TAG) .
 
 gcloudpush: build
-	gcloud docker -- push $(REGISTRY)/kubemark:$(IMAGE_TAG)
+	docker push $(REGISTRY)/kubemark:$(IMAGE_TAG)
 
 push: build
 	docker -- push $(REGISTRY)/kubemark:$(IMAGE_TAG)

--- a/test/e2e_node/conformance/build/Makefile
+++ b/test/e2e_node/conformance/build/Makefile
@@ -76,10 +76,10 @@ endif
 	docker build --pull -t ${IMAGE_NAME}-${ARCH}:${VERSION} ${TEMP_DIR}
 
 push: build
-	gcloud docker -- push ${IMAGE_NAME}-${ARCH}:${VERSION}
+	docker push ${IMAGE_NAME}-${ARCH}:${VERSION}
 ifeq ($(ARCH),amd64)
 	docker tag ${IMAGE_NAME}-${ARCH}:${VERSION} ${IMAGE_NAME}:${VERSION}
-	gcloud docker -- push ${IMAGE_NAME}:${VERSION}
+	docker push ${IMAGE_NAME}:${VERSION}
 endif
 
 .PHONY: all

--- a/test/images/image-util.sh
+++ b/test/images/image-util.sh
@@ -97,7 +97,7 @@ push() {
   fi
   for arch in ${archs}; do
     TAG=$(<${IMAGE}/VERSION)
-    gcloud docker -- push ${REGISTRY}/${IMAGE}-${arch}:${TAG}
+    docker push ${REGISTRY}/${IMAGE}-${arch}:${TAG}
   done
 }
 

--- a/test/images/volumes-tester/ceph/Makefile
+++ b/test/images/volumes-tester/ceph/Makefile
@@ -24,7 +24,7 @@ image:
 	docker tag $(PREFIX)/volume-ceph $(PREFIX)/volume-ceph:$(TAG)  # Add the version tag to the latest image
 
 push: image
-	gcloud docker -- push $(PREFIX)/volume-ceph # Push image tagged as latest to repository
-	gcloud docker -- push $(PREFIX)/volume-ceph:$(TAG) # Push version tagged image to repository (since this image is already pushed it will simply create or update version tag)
+	docker push $(PREFIX)/volume-ceph # Push image tagged as latest to repository
+	docker push $(PREFIX)/volume-ceph:$(TAG) # Push version tagged image to repository (since this image is already pushed it will simply create or update version tag)
 
 clean:

--- a/test/images/volumes-tester/gluster/Makefile
+++ b/test/images/volumes-tester/gluster/Makefile
@@ -24,7 +24,7 @@ image:
 	docker tag $(PREFIX)/volume-gluster $(PREFIX)/volume-gluster:$(TAG)  # Add the version tag to the latest image
 
 push: image
-	gcloud docker -- push $(PREFIX)/volume-gluster # Push image tagged as latest to repository
-	gcloud docker -- push $(PREFIX)/volume-gluster:$(TAG) # Push version tagged image to repository (since this image is already pushed it will simply create or update version tag)
+	docker push $(PREFIX)/volume-gluster # Push image tagged as latest to repository
+	docker push $(PREFIX)/volume-gluster:$(TAG) # Push version tagged image to repository (since this image is already pushed it will simply create or update version tag)
 
 clean:

--- a/test/images/volumes-tester/iscsi/Makefile
+++ b/test/images/volumes-tester/iscsi/Makefile
@@ -34,8 +34,8 @@ block:
 
 push: image
 	# Push image tagged as latest to repository
-	gcloud docker -- push $(PREFIX)/volume-iscsi
+	docker push $(PREFIX)/volume-iscsi
 	# Push version tagged image to repository (since this image is already pushed it will simply create or update version tag)
-	gcloud docker -- push $(PREFIX)/volume-iscsi:$(TAG)
+	docker push $(PREFIX)/volume-iscsi:$(TAG)
 
 clean:

--- a/test/images/volumes-tester/nfs/Makefile
+++ b/test/images/volumes-tester/nfs/Makefile
@@ -24,7 +24,7 @@ image:
 	docker tag $(PREFIX)/volume-nfs $(PREFIX)/volume-nfs:$(TAG) # Add the version tag to the latest image
 
 push: image
-	gcloud docker -- push $(PREFIX)/volume-nfs # Push image tagged as latest to repository
-	gcloud docker -- push $(PREFIX)/volume-nfs:$(TAG) # Push version tagged image to repository (since this image is already pushed it will simply create or update version tag)
+	docker push $(PREFIX)/volume-nfs # Push image tagged as latest to repository
+	docker push $(PREFIX)/volume-nfs:$(TAG) # Push version tagged image to repository (since this image is already pushed it will simply create or update version tag)
 
 clean:

--- a/test/images/volumes-tester/rbd/Makefile
+++ b/test/images/volumes-tester/rbd/Makefile
@@ -34,8 +34,8 @@ block:
 
 push: image
 	# Push image tagged as latest to repository
-	gcloud docker -- push $(PREFIX)/volume-rbd
+	docker push $(PREFIX)/volume-rbd
 	# Push version tagged image to repository (since this image is already pushed it will simply create or update version tag)
-	gcloud docker -- push $(PREFIX)/volume-rbd:$(TAG)
+	docker push $(PREFIX)/volume-rbd:$(TAG)
 
 clean:

--- a/test/images/webhook/Makefile
+++ b/test/images/webhook/Makefile
@@ -17,4 +17,4 @@ build:
 	docker build --no-cache -t gcr.io/kubernetes-e2e-test-images/k8s-sample-admission-webhook-amd64:1.9v1 .
 	rm -rf webhook
 push:
-	gcloud docker -- push gcr.io/kubernetes-e2e-test-images/k8s-sample-admission-webhook-amd64:1.9v1
+	docker push gcr.io/kubernetes-e2e-test-images/k8s-sample-admission-webhook-amd64:1.9v1

--- a/test/kubemark/start-kubemark.sh
+++ b/test/kubemark/start-kubemark.sh
@@ -198,6 +198,10 @@ function create-and-upload-hollow-node-image {
     exit 1
   fi
 
+  echo "Configuring registry authentication"
+  mkdir -p "${HOME}/.docker"
+  gcloud beta auth configure-docker -q
+
   echo "Copying kubemark binary to ${MAKE_DIR}"
   cp "${KUBEMARK_BIN}" "${MAKE_DIR}"
   CURR_DIR=`pwd`
@@ -223,6 +227,10 @@ function create-and-upload-hollow-node-image {
 # Use bazel rule to create a docker image for hollow-node and upload
 # it to the appropriate docker container registry for the cloud provider.
 function create-and-upload-hollow-node-image-bazel {
+  echo "Configuring registry authentication"
+  mkdir -p "${HOME}/.docker"
+  gcloud beta auth configure-docker -q
+
   RETRIES=3
   for attempt in $(seq 1 ${RETRIES}); do
     if ! bazel run //cluster/images/kubemark:push --define REGISTRY="${FULL_REGISTRY}" --define IMAGE_TAG="${KUBEMARK_IMAGE_TAG}"; then

